processing/src/main/java/io/druid/query/aggregation/post/ArithmeticPostAggregator.java

```diff
@@ -71,17 +76,28 @@ public ArithmeticPostAggregator(
   )
   {
     Preconditions.checkArgument(fnName != null, "fn cannot not be null");
-    Preconditions.checkArgument(fields != null && fields.size() > 1, "Illegal number of fields[%s], must be > 1");
 
     this.name = name;
     this.fnName = fnName;
     this.fields = fields;
 
-    this.op = Ops.lookup(fnName);
-    if (op == null) {
-      throw new IAE("Unknown operation[%s], known operations[%s]", fnName, Ops.getFns());
+    this.binary = BinaryOp.lookup(fnName);
+    this.unary = UnaryOp.lookup(fnName);
+
+    if (binary != null) {
+      Preconditions.checkArgument(
+          fields != null && fields.size() >= 2, "Illegal number of fields[%s], must be >= 2"
+      );
+    } else if (unary != null) {
+      Preconditions.checkArgument(
+          fields != null && fields.size() == 1, "Illegal number of fields[%s], must be == 1"
+      );
+    } else {
+      throw new IAE("Unknown operation[%s], known operations[%s and %s] ", fnName, UnaryOp.getFns(), BinaryOp.getFns());
```

Comment:
i have pending PRs for https://github.com/druid-io/druid/pull/2090 and https://github.com/druid-io/druid/pull/1965 for doing this in the most general way. do they not cover the use case of binary/unary op here?